packages/api/src/lib/utils/_body/RequestProxy.ts

```diff
@@ -0,0 +1,159 @@
+	public destination: RequestDestination = '';
+	public integrity: string = '';
+	public keepalive: boolean = false;
+
+	public mode: RequestMode = 'cors';
+	public redirect: RequestRedirect = 'follow';
+	public referrer: string = 'about:client';
+	public referrerPolicy: ReferrerPolicy = '';
+
+	public headers: Headers;
+
+	public bodyUsed: boolean = false;
+	public duplex = 'half' as const;
+
+	readonly #request: ApiRequest;
+	readonly #url: RequestURLProxy;
+	#cachedMethod: MethodName | null = null;
+	#cachedHasBody: boolean | null = null;
+	#cachedBodyArrayBuffer: ArrayBuffer | null = null;
+	#cachedBodyBlob: Blob | null = null;
+	#cachedBodyFormData: FormData | null = null;
+	#cachedBodyJson: unknown = null;
+	#cachedBodyText: string | null = null;
+	#abortController: AbortController | null = null;
+	#bodyStream: ReadableStream<Uint8Array> | null = null;
+
+	public constructor(request: ApiRequest) {
+		this.#request = request;
+		this.#url = new RequestURLProxy(request);
+		this.headers = new RequestHeadersProxy(request);
+	}
+
+	public get url(): string {
+		return this.#url.href;
+	}
+
+	public get method(): MethodName {
+		this.#cachedMethod ??= (this.#request.method?.toUpperCase() ?? 'GET') as MethodName;
+		return this.#cachedMethod;
+	}
+
+	public get signal() {
+		this.#abortController ??= new AbortController();
+		return this.#abortController.signal;
+	}
+
+	public get body(): ReadableStream<Uint8Array> | null {
+		if (!this.hasBody) return null;
+
+		this.#bodyStream ??= new ReadableStream({
+			start: (controller) => {
+				this.#request
+					.on('data', (chunk) => controller.enqueue(chunk))
+					.once('error', (error) => {
+						controller.error(error);
+						this.#abortController?.abort();
+					})
+					.once('close', () => {
+						this.#abortController?.abort();
+					})
+					.once('end', () => {
+						controller.close();
+					});
+			}
+		});
+
+		return this.#bodyStream;
+	}
+
+	public async arrayBuffer(): Promise<ArrayBuffer> {
+		if (this.#cachedBodyArrayBuffer === null) {
+			const { body } = this;
+			if (body === null) {
+				this.#cachedBodyArrayBuffer = new ArrayBuffer(0);
+			} else {
+				const chunks: Uint8Array[] = [];
+				for await (const chunk of body) {
+					chunks.push(chunk);
+				}
+
+				this.#cachedBodyArrayBuffer = Buffer.concat(chunks).buffer;
+			}
+		}
+
+		return this.#cachedBodyArrayBuffer;
+	}
+
+	public async blob(): Promise<Blob> {
+		if (this.#cachedBodyBlob === null) {
+			const arrayBuffer = await this.arrayBuffer();
+			this.#cachedBodyBlob = new Blob([arrayBuffer], {
+				type: this.headers.get('content-type') ?? ''
+			});
+		}
+
+		return this.#cachedBodyBlob;
+	}
+
+	public async formData(): Promise<FormData> {
+		this.#cachedBodyFormData ??= await new Response(this.body, { headers: this.headers }).formData(); // NOSONAR
+		return this.#cachedBodyFormData;
+	}
+
+	public async json(): Promise<unknown> {
+		this.#cachedBodyJson ??= JSON.parse(await this.text());
```

Comment:
Doesn't this technically diverge from actual behavior (which is throw because body was consumed)?